Content.Server/RevolutionFlag/FlagSystem.cs

```diff
@@ -0,0 +1,96 @@
+using Content.Server.RevolutionFlag.Components;
+using Content.Server.Mind.Components;
+using Content.Server.Traitor;
+using Content.Server.Weapon.Melee;
+using Content.Shared.Damage;
+using Content.Shared.Hands;
+using Content.Shared.StatusEffect;
+
+namespace Content.Server.RevolutionFlag;
+    internal sealed class FlagSystem : EntitySystem
+    {
+        [Dependency] private readonly EntityLookupSystem _entityLookup = default!;
+        [Dependency] private readonly StatusEffectsSystem _statusEffectSystem = default!;
+        private const string RevolutionaryPrototypeId = "Revolutionary";
+        private const string CourageEffectKey = "Courage";
+        private DamageModifierSet Modifiers = default!; 
+        
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<FlagBuffComponent,MeleeHitEvent>(ModifyDamage);
+            SubscribeLocalEvent<FlagComponent,GotEquippedHandEvent>(OnEquipped);
+            SubscribeLocalEvent<FlagComponent,GotUnequippedHandEvent>(OnDequipped);
+        }
+
+        private void OnEquipped(EntityUid uid, FlagComponent comp, GotEquippedHandEvent args)
+        {
+            comp.active = true;
+        }
+
+        private void OnDequipped(EntityUid uid, FlagComponent comp, GotUnequippedHandEvent args)
+        {
+            comp.active = false;
+        }
+
+        public void Aura(FlagComponent flagComp, float range)
+        {
+            var transform = EntityManager.GetComponent<TransformComponent>(flagComp.Owner);
+            var mapPosition = transform.MapPosition;
+            var inRange = _entityLookup.GetEntitiesInRange(transform.Coordinates, range);
+
+            if (!flagComp.active)
+                return;
+            foreach(var entityInRange in inRange)
+            {
+                if (!TryComp<MindComponent>(entityInRange, out var mind))
+                    continue;
+                if (!GetRole(mind, RevolutionaryPrototypeId))
+                    continue;
+                _statusEffectSystem.TryAddStatusEffect<FlagBuffComponent>(entityInRange, CourageEffectKey, flagComp.timespan, true);
+            }
+        }
+
+        private void ModifyDamage(EntityUid uid, FlagBuffComponent comp, MeleeHitEvent args)
+        {
+            /* 
+                This is a little bit cheaty but I couldn't 
+                figure out how to specify a damagemodifierset 
+                inside a component to actually modify the damage
+                so for now I'm just adding the base damage to
+                bonus damage
+            */                                                  
+            args.BonusDamage += args.BaseDamage;
+        }
```

Comment:
Revs is already heavily weighted in favour of the crew and they get double damage?

Reply:
I was under the impression that revs was weak, if that's not the case it might be necessary to just scrap the flag as it was mainly meant to be a buff to them

Reply:
Oh I see what your saying, the flagbuffcomp can only ever be applied to revolutionaries, meaning that only revs get the buff not crew